core/arch/arm/plat-ti/sub.mk

```diff
@@ -3,4 +3,4 @@ srcs-y += main.c
 srcs-$(CFG_PL310) += ti_pl310.c
 srcs-$(PLATFORM_FLAVOR_dra7xx) += sm_platform_handler_a15.c
 srcs-$(PLATFORM_FLAVOR_am57xx) += sm_platform_handler_a15.c
-srcs-$(PLATFORM_FLAVOR_am43xx) += sm_platform_handler_a9.c
+srcs-$(PLATFORM_FLAVOR_am43xx) += sm_platform_handler_a9.c a9_plat_init.S ../sm/pm_a32.S ../sm/pm.c
```

Comment:
Are the `sm` files necessary? I would think they are already added, as long as `CFG_ARM_TRUSTED_FIRMARE=n` (see https://github.com/OP-TEE/optee_os/blob/59ac3927b3bd14e133f57e7aa59278b09c2ca95d/core/arch/arm/arm.mk#L57-L59).

Reply:
Ignore my previous comment, I did not read the file names correctly.

If you need to use only `../sm/pm_a32.S` and `../sm/pm.c`, you need to split the following line in `core/arch/arm/sm/sub.mk` and have them included by some `CFG_` variable:
```
srcs-$(CFG_PSCI_ARM32) += std_smc.c psci.c pm.c psci-helper.S pm_a32.S
```
Indeed, it is not good practice to include files from an external directory.

Reply:
I didn't like doing that either, but I wasn't sure how to split up CFG_PSCI_ARM32, so was looking for guidance on that. Maybe:
```
srcs-$(CFG_PM_ARM32) += pm.c pm_a32.S
srcs-$(CFG_PSCI_ARM32) += std_smc.c psci.c psci-helper.S
```
But then how do we enforce the dependency from PSCI -> PM here without Kconfig, some kind of rule in a .mk file?

Reply:
This should be enough:
```make
ifeq ($(CFG_PSCI_ARM32),y)
$(call force,CFG_PM_ARM32,y)
endif
```
Since this pattern appears in several places, we could as well write a function to declare dependencies, but I'm a bit worried we may end up writing something uselessly complex...